public/scripts/s3-uploads.js

```diff
@@ -1,61 +1,91 @@
-(function () {
-  document.getElementById('camera-input').onchange = function () {
-    const files = document.getElementById('camera-input').files;
-    const file = files[0];
-    if (file == null) {
-      return new Error('No file selected.');
-    }
-    getSignedRequest(file);
-  };
-}());
-
-let imageUploadPending = false;
 
-function getSignedRequest(file) {
-  const xhr = new XMLHttpRequest();
-  xhr.onreadystatechange = () => {
-    if (xhr.readyState === 4) {
-      if (xhr.status === 200) {
-        const response = JSON.parse(xhr.responseText);
-        uploadFile(file, response.signedRequest, response.url);
-        imageUploadPending = true;
-        updateTagAndHeading(response.imageId);
+function s3handling(fileSource) {
+  (function () {
```

Comment:
This iife is nonsense.

You need to use require / module.exports. This is no no good we are going backwards!